js/background.js

```diff
@@ -19,47 +19,48 @@ function handleFirstRun(details) {
 //
 // Stores a request if we haven't seen it before; otherwise updates it.
 //
-function requestUpdated(info) {
+async function requestUpdated(info) {
     if (Math.sign(info.tabId) >= 0) {
-        chrome.tabs.get(info.tabId, function (tab) {
-            if (tab) {
-                if (tab.url) {
-                    for (var i = 0; i < IGNORED_PREFIXES.length; i++) {
-                        if (tab.url.indexOf(IGNORED_PREFIXES[i]) === 0) {
-                            return;
-                        }
+        try {
+            const tab = await browser.tabs.get(info.tabId);
+            if (tab.url) {
+                for (var i = 0; i < IGNORED_PREFIXES.length; i++) {
+                    if (tab.url.indexOf(IGNORED_PREFIXES[i]) === 0) {
+                        return;
                     }
-                } else if (info.url === WORKBENCH_URL) {
-                    return;
                 }
+            } else if (info.url === WORKBENCH_URL) {
+                return;
+            }
 
-                // Track the request start time if it's a new request
-                if (!requests[info.requestId]) {
-                    requests[info.requestId] = {
-                        timeStarted: new Date().getTime()
-                    };
-                }
+            // Track the request start time if it's a new request
+            if (!requests[info.requestId]) {
+                requests[info.requestId] = {
+                    timeStarted: new Date().getTime()
+                };
+            }
 
-                // Base64 encode the body parts if necessary
-                if (info.requestBody && info.requestBody.raw) {
-                    for (var i = 0; i < info.requestBody.raw.length; i++) {
-                        var part = info.requestBody.raw[i];
+            // Base64 encode the body parts if necessary
+            if (info.requestBody && info.requestBody.raw) {
+                for (var i = 0; i < info.requestBody.raw.length; i++) {
+                    var part = info.requestBody.raw[i];
 
-                        if (part.bytes) {
-                            part.base64 = toBase64(part.bytes);
-                        } else if (part.file) {
-                        }
+                    if (part.bytes) {
+                        part.base64 = toBase64(part.bytes);
+                    } else if (part.file) {
                     }
                 }
+            }
 
-                // Copy properties
-                for (var prop in info) {
-                    if (info.hasOwnProperty(prop)) {
-                        requests[info.requestId][prop] = info[prop];
-                    }
+            // Copy properties
+            for (var prop in info) {
+                if (info.hasOwnProperty(prop)) {
+                    requests[info.requestId][prop] = info[prop];
                 }
             }
-        })
+        } catch (err) {
+            console.error(err);
```

Comment:
Are there specific errors that you've observed here, or just catching them in case there are any?

Reply:
yes, nothing specific. I can remove that if you like

Reply:
Nah, may as well keep it. I was just curious.